src/app/wallets/get-transactions-for-wallet.ts

```diff
@@ -0,0 +1,61 @@
+import { RepositoryError } from "@domain/errors"
+import { OnChainError, MakeTxFilter, MakeTxDecoder } from "@domain/bitcoin/onchain"
+import { MakeWalletsRepository } from "@services/mongoose"
+import { MakeLedgerService } from "@services/ledger"
+import { MakeOnChainService } from "@services/lnd/onchain-service"
+import { toLiabilitiesAccountId, LedgerError } from "@domain/ledger"
+import { LOOK_BACK } from "@core/utils"
+import { ONCHAIN_MIN_CONFIRMATIONS } from "@config/app"
+import { WalletTransactionHistory } from "@domain/wallets"
+
+// TODO should be exposed via PriceSerivce / LiquidityProvider
+import { getCurrentPrice } from "@services/realtime-price"
+
+export const getTransactionsForWalletId = async ({
+  walletId,
+}: {
+  walletId: WalletId
+}): Promise<[WalletTransaction[], ApplicationError?]> => {
```

Comment:
I really dont understand why we are over complicating this, just return transactions because we are throwing an error to the user if we have problems with incoming txs

Reply:
Right now we are throwing an error to move forward but we want to translate and return the error properly via graphql